data-model/src/main/resources/META-INF/native-image/io.micronaut.data/data-model/native-image.properties

```diff
@@ -14,4 +14,8 @@
 # limitations under the License.
 #
 
-Args = --initialize-at-build-time=io.micronaut.data.model.query.builder.sql.Dialect,io.micronaut.data.model.DataType,io.micronaut.data.annotation.Join$Type
+Args = --initialize-at-build-time=io.micronaut.data.model.query.builder.sql.Dialect \
+  --initialize-at-build-time=io.micronaut.data.model.DataType \
+  --initialize-at-build-time=io.micronaut.data.annotation.Join$Type \
+  --initialize-at-build-time=io.micronaut.data.model.JsonDataType \
+  --initialize-at-build-time=io.micronaut.data.intercept.annotation.DataMethod$OperationType
```

Comment:
I had to add these two build time initializations in a pet project 

https://github.com/questionnaire-training/questionnaire-training/commit/ad5ef17e9e4511e105cdc1ac7e7c5d01a4011552